fix(api,hardware-testing): Fix wrong polarity in module offsets + refactor module calibration test script and add verification step.

vegano1 opened the pull request:
# Overview

The calibrated module offsets had the wrong polarity and needed to be flipped for deck items, we also needed a way to verify the calibrated square center was correct when running the calibration testing script. The verification step takes the calibrated well A1 (square center) and removes the previously calibrated offset to visually compare the calibrated/uncalibrated positions.

Closes [RCORE-739](https://opentrons.atlassian.net/browse/RCORE-739)

# Test Plan

- [x] Make sure we can calibrate modules using the HTTP script
- [x] Make sure we can verify the calibrated offset using the HTTP script
- [x] Make sure the calibrated module offset is saved with the correct polarity
- [x] Make sure instrument calibrations still work as expected and the offset is saved correctly 

# Changelog

- Fixed issue with module calibration offset getting generated with the wrong polarity
- Added calibration_target arg and enums to `find_calibration_structure_position` method so we can reverse the calibration offset based on whether we are calibrating a deck object (module) or gantry instrument (pipette, gripper, etc)
- Refactor the module calibration test script and added OpentronsHTTPAPI HTTP wrapper class to consolidate functionality
- Added a verification step to the module calibration test script so we can confirm the center square with/without offsets.
- Added `urllib3, certifi, and charset-normalizer` to Pipfile.lock for hardware_testing module


#  Review requests

- Make sure this works as intended
- Is the verification step sufficient for what we need?
- Anything else I'm missing?

# Risk assessment
Low, flex work


[RCORE-739]: https://opentrons.atlassian.net/browse/RCORE-739?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

vegano1 committed:
fix(api,hardware-testing): Fix wrong polarity in module offsets + refactor module calibration test script and add verification step.

sfoster1 commented on hardware-testing/hardware_testing/scripts/module_calibration.py:
```diff
@@ -31,161 +36,335 @@
-    url = f"{BASE_URL.format(ip_addr)}/runs"
-    res = requests.post(headers=HEADERS, url=url)
-    if res.status_code != 201:
+class OpentronsHTTPAPI:
```
can we drop this class in its own module somewhere?

sfoster1 commented on api/src/opentrons/hardware_control/ot3_calibration.py:
```diff
@@ -784,6 +784,9 @@ async def calibrate_module(
             offset = await find_calibration_structure_position(
```
would love to see the offset come out of this method correct by passing like `calibration_target=CalibrationTarget.DECK_OBJECT` (as opposed to `CalibrationTarget.GANTRY_INSTRUMENT` or something which would be the default)

sfoster1 commented on api/src/opentrons/hardware_control/ot3_calibration.py:
```diff
@@ -784,6 +784,9 @@ async def calibrate_module(
                 hcapi, mount, nominal_position, method=CalibrationMethod.BINARY_SEARCH
             )
+
+            # need to flip the polarity since modules are fixed on the deck
```
I think it's worth expanding on this a little, it makes sense if you know the full context but maybe not to someone looking at this code in 5 years.

sfoster1 reviewed (approved):
Looks good overall! There are some little refactors that I think would be really useful but it could merge without them.

vegano1 committed:
move OpentronsHTTPAPI class to new http_api module so its reusable
use Point class from hardware project instead of recreating it
add CalibrationTarget enum and use it to return the correct calibration offset polarity in find_calibration_structure_position
add urllib3, certifi, and charset-normalizer to Pipfile.lock for hardware_testing module

vegano1 committed:
fix linting issues

vegano1 committed:
handle heater-shaker and thermocycler latch/lids

sfoster1 reviewed (approved):
Looks good! Love the new api class.

vegano1 committed:
fix linting issues
